File: graph_search.py
graph = {
    'D' : ['J', 'G', 'F'],
    'J' : ['D'],
    'G' : ['D', 'E', 'H'],
    'H' : ['I', 'G'],
    'I' : ['H'],
    'E' : ['G'],
    'F' : ['M', 'K', 'D'],
    'M' : ['F'],
    'K' : ['F']
}

visited_dfs = []
visited_bfs = []
queue = []


# Depth-first search
def dfs(graph, node):
    global visited_dfs
    # if not visited
    if node not in visited_dfs:
        visited_dfs.append(node)
        # visiting other nodes
        for n in graph[node]:
            dfs(graph, n)
            
    
# Breadth-first search        
def bfs(graph, node):
    visited_bfs.append(node)
    queue.append(node)
    
    # while queue not empty
    while queue:
        s = queue.pop(0)
        
        # for every branch 
        for neighbor in graph[s]:
            if neighbor not in visited_bfs:
                visited_bfs.append(neighbor)
                queue.append(neighbor)
            
                   
dfs(graph, 'D')
print(f"Path in depth from vertex D: {visited_dfs}")

bfs(graph, 'F')
print(f"Path in width from vertex F: {visited_bfs}")

File: graph_search.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;
import java.util.Queue;

public class graph_search {

    private static Map<Character, char[]> Graph = new HashMap<Character, char[]>();
    private static List<Character> visited_dfs = new ArrayList<Character>();
    private static List<Character> visited_bfs = new ArrayList<Character>();
    private static Queue<Character> queue = new LinkedList<>();
    public static void main(String[] args) {
        Graph.put('D', new char[] {'J', 'G', 'F'});
        Graph.put('J', new char[] {'D'});
        Graph.put('G', new char[] {'D', 'E', 'H'});
        Graph.put('H', new char[] {'I', 'G'});
        Graph.put('I', new char[] {'H'});
        Graph.put('E', new char[] {'G'});
        Graph.put('F', new char[] {'M', 'K', 'D'});
        Graph.put('M', new char[] {'F'});
        Graph.put('K', new char[] {'F'});

        dfs(Graph, 'D');
        System.out.println(String.format("Path in depth from vertex D: %s", visited_dfs));

        bfs(Graph, 'F');
        System.out.println(String.format("Path in width from vertex F: %s", visited_bfs));
    };

    // Depth-first search
    private static void dfs(Map<Character, char[]> graph, char node) {
        // if not visited
        if (!visited_dfs.contains(node)) {
            visited_dfs.add(node);
            // visiting other nodes
            for (char neighbor : Graph.get(node)) {
                dfs(Graph, neighbor);    
            }
        }
    }

    // Breadth-first search
    private static void bfs(Map<Character, char[]> graph, char node) {
        visited_bfs.add(node);
        queue.add(node);

        // while queue not empty
        while (queue.size() != 0) {
            char s = queue.remove();

            // for every branch
            for (char neighbor : Graph.get(s)) {
                if (!visited_bfs.contains(neighbor)) {
                    visited_bfs.add(neighbor);
                    queue.add(neighbor);
                }
            }
        }
    }
}